src/nunavut/lang/c/__init__.py

```diff
@@ -94,6 +94,9 @@ def strop(self,
               token: str,
               reserved_words: typing.FrozenSet[str],
               reserved_patterns: typing.Optional[typing.FrozenSet[typing.Pattern]] = None) -> str:
+        if len(token) == 0:
+            return token
```

Comment:
So I'm not sure if this is "correct", this was just the easiest way to address the problems I was causing with my usage of `filter_id`.

Essentially at some point I was passing an empty string through to filter, which translates that into an underscore. This just returns an empty string. Putting no real critical thinking into it I think this is more correct, I wouldn't want an empty string to be expanded but I don't know if thats a valid way of thinking about it.

Reply:
An empty identifier is not a valid identifier so the correct behavior would be to raise an error. Maybe you could look into your usage of filter_id to make sure it's never invoked on an empty string?

Reply:
I was afraid you'd say that :(.

I know where it's being passed an empty string, it's just going to need a bit of rip-up and tweaking there.

Reply:
such is life David

Reply:
With stropping enabled (which it is by default) the empty string, being an invalid identifier, is transformed into a valid identifier. This actually corresponds nicely with the Python convention of using a single underscore as a "throw-away" variable.